pkg/server/metrics/metrics.go

```diff
@@ -134,3 +154,13 @@ func (a *ServerMetrics) ConnectionDec(serviceMethod string) {
 func (a *ServerMetrics) SetBackendCount(count int) {
 	a.backend.WithLabelValues().Set(float64(count))
 }
+
+// PendingDialInc increments a pending agent dial.
```

Comment:
I think its safer to set the count then trust to always get the increment/decrement called.

Reply:
ack